### DISCUSSION TITLE
One time private application (no sharing)

### DISCUSSION THREAD

**Discussion by ghost** [netblue30/firejail#5047] [General]
I am sorry for asking this here but I don't know where else I could have.

I am interested in using firejail for creating a disposable sandbox for a browser.

The thing is, while that instance is running it should not allow anyone to access its contents by running a command.

It should be restricted to the window that it created in the first place.

This is for security reasons.

Is this something that firejail supports and if any special parameters are required, what would those be?

**Comment by smitsohu** [comment]
Firejail creates a security boundary between inside the sandbox and outside the sandbox.

Code running inside the sandbox should not be able to do things that we didn't allow it to do. Firejail pretty much doesn't care about the world outside the sandbox. What you want to do is not part of the job description of a sandbox.

It is not impossible that a sandbox can help to achieve your goals, for example by blocking D-Bus and other IPC channels, as for IPC it doesn't matter which end you block. It might make sense to use a new network namespace (`--net`). But Firejail and similar tools will always fall short of a comprehensive solution.

What I would try in your place is running the browser as a different user in a Wayland session, which should solve most of the obvious problems. Check `/proc/sys/kernel/yama/ptrace_scope`, it should exist and be set to `1` or higher (to defend the Wayland server). Even then a talented adversary might find workarounds. I don't think this is an easy task.

**Comment by ghost** [comment]
@smitsohu Thank you for the response, much appreciated. Could you please give a few more details of the plan that you suggested that involves Wayland? I am an xorg ubuntu 20.04 user.

**Reply by smitsohu** [reply]
Every process in the X11 session can create screenshots and log and inject key strokes. Probably this is not what you want, and is solved by using Wayland instead.

Then you probably don't want to allow other processes to play with your browser settings, read browser caches, `ptrace` the browser or send it signals. A separate user just for the browser should solve _these_ problems.

In order to allow the new user access to your Wayland session, you need to give it search/read/write permission on the Wayland socket, which is `/run/user/$UID/wayland-0` or `$XDG_RUNTIME_DIR/wayland-0` usually. Then you need to point environment variable `WAYLAND_DISPLAY` to this socket. 

I think something like this should work, but I didn't try myself, so YMMV.

~~(now that I'm writing this down I'm starting to wonder if the combination of a different user and Firejail's `--x11=xpra` option could be an alternative to Wayland).~~
